helix-core/src/main/java/org/apache/helix/spectator/RoutingTableProvider.java

```diff
@@ -355,7 +565,20 @@ public Object removeRoutingTableChangeListener(
    * @return empty list if there is no instance in a given state
    */
   public Set<InstanceConfig> getInstancesForResourceGroup(String resourceGroupName, String state) {
-    return _routingTableRef.get().getInstancesForResourceGroup(resourceGroupName, state);
+    String key = getRoutingTableKey();
+    return _routingTableRef.get(key).get().getInstancesForResourceGroup(resourceGroupName, state);
+  }
+
+  public Set<InstanceConfig> getInstancesForResourceGroup(String resourceGroupName, String state,
```

Comment:
Same here. If not in the design, let's hold this for now.

Reply:
Removed.